flash/image/classification/adapters.py

```diff
@@ -0,0 +1,416 @@
+from flash.core.data.auto_dataset import BaseAutoDataset
+from flash.core.data.data_source import DefaultDataKeys
+from flash.core.model import Task
+from flash.core.registry import FlashRegistry
+from flash.core.utilities.imports import _LEARN2LEARN_AVAILABLE
+from flash.core.utilities.providers import _LEARN2LEARN
+from flash.core.utilities.url_error import catch_url_error
+
+if _LEARN2LEARN_AVAILABLE:
+    import learn2learn as l2l
+
+    class RemapLabels(l2l.data.transforms.TaskTransform):
+        def __init__(self, dataset, shuffle=True):
+            super().__init__(dataset)
+            self.dataset = dataset
+            self.shuffle = shuffle
+
+        def remap(self, data, mapping):
+            data[DefaultDataKeys.TARGET] = mapping(data[DefaultDataKeys.TARGET])
+            return data
+
+        def __call__(self, task_description):
+            if task_description is None:
+                task_description = self.new_task()
+            labels = list({self.dataset.indices_to_labels[dd.index] for dd in task_description})
+            if self.shuffle:
+                random.shuffle(labels)
+
+            def mapping(x):
+                return labels.index(x)
+
+            for dd in task_description:
+                remap = functools.partial(self.remap, mapping=mapping)
+                dd.transforms.append(remap)
+            return task_description
+
+
+class NoModule:
+
+    """This class is used to prevent nn.Module infinite recursion."""
+
+    def __init__(self, task):
+        self.task = task
+
+    def __getattr__(self, key):
+        if key != "task":
+            return getattr(self.task, key)
+        return self.task
+
+    def __setattr__(self, key: str, value: Any) -> None:
+        if key == "task":
+            object.__setattr__(self, key, value)
+            return
+        setattr(self.task, key, value)
+
+
+class Epochifier:
+    def __init__(self, tasks, length):
+        self.tasks = tasks
+        self.length = length
+
+    def __getitem__(self, *args, **kwargs):
+        return self.tasks.sample()
+
+    def __len__(self):
+        return self.length
+
+
+class Model(torch.nn.Module):
+    def __init__(self, backbone: torch.nn.Module, head: Optional[torch.nn.Module]):
+        super().__init__()
+        self.backbone = backbone
+        self.head = head
+
+    def forward(self, x):
+        x = self.backbone(x)
+        if x.dim() == 4:
+            x = x.mean(-1).mean(-1)
+        if self.head is None:
+            return x
+        return self.head(x)
+
+
+class Learn2LearnAdapter(Adapter):
+    """The ``Learn2LearnAdapter`` is an :class:`~flash.core.adapter.Adapter` for integrating with learn 2 learn
+    library."""
+
+    required_extras: str = "image"
+
+    def __init__(
+        self,
+        task: AdapterTask,
+        backbone: torch.nn.Module,
+        head: torch.nn.Module,
+        algorithm_cls: Type[LightningModule],
+        ways: int,
+        kshots: int,
+        queries: int = 1,
```

Comment:
Could do with some docstrings here, I'm not sure what all these are haha

Reply:
Done !